### PR TITLE
Use XDG_RUNTIME_DIR for storing Xauthority

### PR DESCRIPTION
**Pull request by UtkarshVerma**
This PR saves `lxauth` in `XDG_RUNTIME_DIR`.

### CODE REVIEW

File: src/login.c [hunk 1]
```diff
@@ -304,10 +304,10 @@ void remove_utmp_entry(struct utmp *entry) {
 	endutent();
 }
 
-void xauth(const char* display_name, const char* shell, const char* dir)
+void xauth(const char* display_name, const char* shell)
 {
 	char xauthority[256];
-	snprintf(xauthority, 256, "%s/%s", dir, ".lyxauth");
+	snprintf(xauthority, 256, "%s/%s", getenv("XDG_RUNTIME_DIR"), "lyxauth");
```
Comment (nullgemm) [inline src/login.c:310]: We have to test whether this var is empty or not! If it is, we could try using `XDG_CONFIG_HOME` instead, and if it also is empty, we can use `pwd->pw_dir` as a last resort.

File: src/login.c [hunk 2]
```diff
@@ -343,18 +343,10 @@ void xorg(
 	const char* vt,
 	const char* desktop_cmd)
 {
-	// generate xauthority file
-	const char* xauth_dir = getenv("XDG_CONFIG_HOME");
-
-	if ((xauth_dir == NULL) || (*xauth_dir == '\0'))
-	{
-		xauth_dir = pwd->pw_dir;
-	}
```
Comment (nullgemm) [inline src/login.c:352]: This is the kind of test we should be using

**Review by nullgemm — COMMENTED** [review]
Cool, I have made a few comment, once everything is ready this is definitely something I can merge :)

**Comment by UtkarshVerma** [comment]
Implemented your suggestions in the last commit. Please have a look.

**Pull request merged**